### PR TITLE
New features! Bugfixes!

### PR DESCRIPTION
**Pull request by otakup0pe**
* The  `diff` action will show you what is expected to change. This addresses https://github.com/Autodesk/aomi/issues/70
* You can now `export` non write-only secrets from the Vault server. This addresses https://github.com/Autodesk/aomi/issues/92
* You can now `render` the Secretfile and accoutrement without actually interacting with the Vault server
* Can now write to the cubbyhole with generic secrets. Addresses https://github.com/Autodesk/aomi/issues/108
* As part of the cubbyhole bit I wrapped the `hvac.Client` class. Our variant still keeps the fancy metadata annotated "operational" token, but will use the initial token when interacting with a cubbyhole. Other than helping fix https://github.com/Autodesk/aomi/issues/108 this opens up the potential for cubbyhole caching of things like AWS secrets.

### CODE REVIEW

**Comment by otakup0pe** [comment]
cc @inferiorhumanorgans

File: aomi/model/auth.py [hunk 1]
```diff
@@ -86,6 +97,18 @@ class AppRole(Auth):
     required_fields = ['name', 'policies']
     config_key = 'approles'
 
+    @staticmethod
```
Comment (inferiorhumanorgans) [inline aomi/model/auth.py]: https://stackoverflow.com/questions/38987/how-to-merge-two-python-dictionaries-in-a-single-expression#answer-26853961

File: aomi/cli.py [hunk 2]
```diff
@@ -320,45 +368,56 @@ def template_runner(client, parser, args):
 
 def ux_actions(parser, args):
     """Handle some human triggers actions"""
-    if args.operation == 'help':
-        help_me(parser, args)
-
     # cryptorito uses native logging (as aomi should tbh)
-
     normal_fmt = '%(message)s'
-    if args.verbose == 2:
```
Comment (inferiorhumanorgans) [inline aomi/cli.py]: Can this be simplified and/or turned into a switch statement?

Comment (otakup0pe) [inline aomi/cli.py]: Kind of? It now at least doens't throw errors if you aren't asking for verbosity...

File: aomi/model/auth.py (hunk 1 above)
Comment (otakup0pe) [inline aomi/model/auth.py]: This isn't actually for merging tho.

**Pull request merged**